substrate/primitives/src/lib.rs

```diff
@@ -143,5 +143,14 @@ impl AsRef<[u8]> for Data {
   }
 }
 
+/// Lexicographically reverses a given byte array.
+pub fn reverse_lexicographic_order<const N: usize>(bytes: [u8; N]) -> [u8; N] {
```

Comment:
Let's take a moment and add a test this works for all 2-byte arrays?

Re-reviewing this, I'm not sure it's wrong, yet I'm also not sure I wrote the right algorithm. Substrate uses a nibble-based solution and this is a bit/byte-based. It may be simply as nibble order may be allowed to be consistent? Yet easiest to exhaustively test and call it a day.

Just populate two StorageMaps (value, reverse(value)) -> value and check they're as expected.